pandas/core/index.py

```diff
@@ -2119,11 +2119,27 @@ def _maybe_cast_slice_bound(self, label, side):
         label : object
         side : {'left', 'right'}
 
+        Returns
+        -------
+        label :  object
+
         Notes
         -----
         Value of `side` parameter should be validated in caller.
 
         """
+
+        # pass thru float indexers if we have a numeric type index
+        # which then can decide to process / or convert and warng
+        if is_float(label):
+            if not self.is_floating():
+                raise TypeError('Cannot index type {0} with float keys'.format(type(self)))
```

Comment:
if you try this:

```
s = Series(range(5))
s.loc[1.5:5.5]
```

if needs to intercept this and raise rather than actually working (as you are using floats with a non-float index). you can of course use floats/ints with a float index.